pkg/controller/externalsecrets_controller.go

```diff
@@ -40,23 +74,295 @@ type ExternalSecretsReconciler struct {
+		WatchesMetadata(&corev1.Secret{}, handler.EnqueueRequestsFromMapFunc(mapFunc), controllerWatchResourcePredicates).
+		WatchesMetadata(&webhook.ValidatingWebhookConfiguration{}, handler.EnqueueRequestsFromMapFunc(mapFunc), controllerWatchResourcePredicates).
 		Complete(r)
 }
+
+func (r *ExternalSecretsReconciler) processReconcileRequest(externalsecrets *operatorv1alpha1.ExternalSecretsOperator, req types.NamespacedName) (ctrl.Result, error) {
+	createRecon := false
+	if !containsProcessedAnnotation(externalsecrets) && reflect.DeepEqual(externalsecrets.Status, operatorv1alpha1.ExternalSecretsStatus{}) {
+		r.log.V(1).Info("starting reconciliation of newly created external-secrets", "namespace", externalsecrets.GetNamespace(), "name", externalsecrets.GetName())
+		createRecon = true
+	}
+
+	//if err := r.disallowMultipleIstioCSRInstances(externalsecrets); err != nil {
+	//	if IsMultipleInstanceError(err) {
+	//		r.eventRecorder.Eventf(externalsecrets, corev1.EventTypeWarning, "MultiIstioCSRInstance", "creation of multiple istiocsr instances is not supported, will not be processed")
+	//		err = nil
+	//	}
+	//	return ctrl.Result{}, err
+	//}
+
+	var errUpdate error = nil
+	if err := r.reconcileExternalSecretsDeployment(externalsecrets, createRecon); err != nil {
+		r.log.Error(err, "failed to reconcile external-secrets deployment", "request", req)
+		if IsIrrecoverableError(err) {
+			if externalsecrets.Status.SetCondition(operatorv1alpha1.Degraded, metav1.ConditionTrue, operatorv1alpha1.ReasonFailed, fmt.Sprintf("reconciliation failed with irrecoverable eror not retrying: %v", err)) ||
+				externalsecrets.Status.SetCondition(operatorv1alpha1.Ready, metav1.ConditionFalse, operatorv1alpha1.ReasonReady, "") {
+				errUpdate = r.updateCondition(externalsecrets, nil)
+			}
+			return ctrl.Result{}, errUpdate
+		} else {
+			if externalsecrets.Status.SetCondition(operatorv1alpha1.Degraded, metav1.ConditionFalse, operatorv1alpha1.ReasonReady, "") ||
+				externalsecrets.Status.SetCondition(operatorv1alpha1.Ready, metav1.ConditionFalse, operatorv1alpha1.ReasonInProgress, fmt.Sprintf("reconciliation failed, retrying: %v", err)) {
+				errUpdate = r.updateCondition(externalsecrets, err)
+			}
+			return ctrl.Result{RequeueAfter: defaultRequeueTime}, fmt.Errorf("failed to reconcile %q external-secrets deployment: %w", req, errUpdate)
+		}
+	}
+
+	if externalsecrets.Status.SetCondition(operatorv1alpha1.Degraded, metav1.ConditionFalse, operatorv1alpha1.ReasonReady, "") ||
+		externalsecrets.Status.SetCondition(operatorv1alpha1.Ready, metav1.ConditionTrue, operatorv1alpha1.ReasonReady, "reconciliation successful") {
+		errUpdate = r.updateCondition(externalsecrets, nil)
+	}
+	return ctrl.Result{}, errUpdate
+}
+
+// cleanUp handles deletion of external-secrets.openshift.operator.io gracefully.
+func (r *ExternalSecretsReconciler) cleanUp(externalsecrets *operatorv1alpha1.ExternalSecretsOperator) (bool, error) {
```

Comment:
sounds eerie,
the func doc says it is attempting to perform "graceful deletion" but we're just logging a Warning event without performing any action.